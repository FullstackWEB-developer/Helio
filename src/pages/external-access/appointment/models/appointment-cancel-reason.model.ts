export interface AppointmentCancelReason {
    appointmentCancelReasonId: number;
    name: string;
    noShow: boolean;
    patientRescheduled: boolean;
    providerUnavailable: boolean;
    slotAvailable: boolean;
}
