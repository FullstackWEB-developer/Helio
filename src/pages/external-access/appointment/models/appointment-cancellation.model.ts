export interface AppointmentCancellationModel {
    patientId: number;
    appointmentCancelReasonId: number;
    cancellationReason: string;
}
