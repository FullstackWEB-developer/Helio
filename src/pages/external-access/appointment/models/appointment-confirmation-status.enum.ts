export enum ConfirmationStatus {
    ConfirmedViaSMS = 161,
    ConfirmedViaEmail = 121
}