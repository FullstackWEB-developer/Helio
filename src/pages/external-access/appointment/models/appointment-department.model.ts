export interface AppointmentDepartmentModel {
    latLong: number[];
    name: string;
    address: string;
    address2?: string;
    city: string;
    state: string;
    zip: string;
}
