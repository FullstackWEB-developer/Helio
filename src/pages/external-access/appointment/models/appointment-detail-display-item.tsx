export interface AppointmentDetailDisplayItem {
    title: string,
    description: string
}
