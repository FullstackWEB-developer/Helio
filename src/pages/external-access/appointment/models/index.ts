export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
