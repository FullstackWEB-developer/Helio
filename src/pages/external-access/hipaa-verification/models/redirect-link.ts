import {ExternalAccessRequestTypes} from '@pages/external-access/models/external-updates-request-types.enum';
import {RequestChannel} from '@pages/external-access/hipaa-verification/models/request-channel.enum';

export interface RedirectLink {
    attributes: object,
    fullUrl: string,
    linkCreationDate: Date
    linkId: string
    patientId: string
    requestChannel: RequestChannel
    requestType: ExternalAccessRequestTypes
    ticketId: string
}
