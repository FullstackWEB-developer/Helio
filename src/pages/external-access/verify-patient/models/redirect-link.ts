import {ExternalAccessRequestTypes} from '@pages/external-access/models/external-updates-request-types.enum';
import {RequestChannel} from '@pages/external-access/verify-patient/models/request-channel.enum';

export interface RedirectLink {
    attributes: any;
    fullUrl: string;
    linkCreationDate: Date;
    linkId: string;
    patientId: string;
    requestChannel: RequestChannel;
    requestType: ExternalAccessRequestTypes;
    ticketId: string;
    sentAddress: string;
}
