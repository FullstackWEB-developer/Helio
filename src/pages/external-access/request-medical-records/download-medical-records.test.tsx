import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, fireEvent, render, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import DownloadMedicalRecords from './download-medical-records';
import { VerifiedPatient } from '@pages/patients/models/verified-patient';
import { RedirectLink } from '../verify-patient/models/redirect-link';
import { RequestChannel } from '../verify-patient/models/request-channel.enum';
import { ExternalAccessRequestTypes } from '../models/external-updates-request-types.enum';
import { DownloadMedicalRecordsProps } from '@pages/patients/services/patients.service';
import Api from '@shared/services/api';
jest.setTimeout(70000);
describe("DownloadMedicalRecords tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        },
        patientsState: {
            verifiedPatient: {
                firstName: "string",
                lastName: "string",
                patientId: 123,
                primaryProviderId: 123,
                departmentId: 123,
                defaultProviderId: 123,
                defaultDepartmentId: 123,
                dateOfBirth: new Date("10/10/1995"),
                token: "string",
                tokenExpiration: new Date("10/10/1995")
            } as VerifiedPatient
        },
        externalAccessState: {
            verifyPatientState: {
                redirectLink: {
                    attributes: {},
                    fullUrl: "string",
                    linkCreationDate: new Date("10/10/1995"),
                    linkId: "string",
                    patientId: "string",
                    requestChannel: RequestChannel.Chat,
                    requestType: ExternalAccessRequestTypes.AppointmentConfirmation,
                    ticketId: "string",
                    sentAddress: "string",
                } as RedirectLink
            },
            requestMedicalRecordState: {
                data: {
                    patientId: 123,
                    departmentId: 12,
                    downloadLink: "string",
                    emailAddress: "string",
                    isDownload: true,
                    startDate: new Date("10/10/2007"),
                    endDate: new Date("10/10/2007"),
                    asHtml: true,
                    note: "string",
                } as DownloadMedicalRecordsProps
            }
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders DownloadMedicalRecords correctly", async () => {
        let fragment: any;
        jest.spyOn(Api, 'get').mockResolvedValue(true);
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                    <DownloadMedicalRecords/>
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})
