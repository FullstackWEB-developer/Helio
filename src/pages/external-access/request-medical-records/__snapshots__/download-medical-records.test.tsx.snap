// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadMedicalRecords tests renders DownloadMedicalRecords correctly 1`] = `
<DocumentFragment>
  <div
    class="md:px-12 xl:px-48 without-default-padding pt-4 xl:pt-16"
  >
    <div
      class="md:whitespace-pre md:h-24 my-1 md:pb-10 w-full items-center"
    >
      <h4>
        external_access.download_medical_records.title
      </h4>
    </div>
    <div
      class="pt-10 pb-6 xl:pt-2 xl:pb-10"
    >
      external_access.download_medical_records.verify_desc
    </div>
    <div>
      <form>
        <div
          class="pb-6"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  w-full md:w-88"
            >
              <input
                autocomplete="off"
                class="flex-1 flex-auto flex-grow flex-shrink w-px pt-6 pl-4 leading-normal body2 h-14 input-date-mask date-label-mask"
                placeholder=""
                spellcheck="false"
                type="tel"
                value=""
              />
              <label
                class="absolute truncate required body2 "
              >
                external_access.download_medical_records.date_of_birth
              </label>
            </div>
            <div
              class="h-6 pl-4 body3 pt-1 truncate "
            >
              mm/dd/yyyy
            </div>
          </div>
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  w-full md:w-88"
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="phoneNumber"
                data-testid="phoneNumber"
                name="phoneNumber"
                placeholder=""
                required=""
                type="tel"
                value=""
              />
              <label
                class="absolute truncate required body2 "
              >
                external_access.download_medical_records.mobile_phone_number
              </label>
            </div>
          </div>
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  w-full md:w-88"
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="zip"
                data-testid="zip"
                maxlength="5"
                name="zip"
                placeholder=""
                required=""
                type="zip"
                value=""
              />
              <label
                class="absolute truncate required body2 "
              >
                external_access.download_medical_records.zip_code
              </label>
            </div>
          </div>
        </div>
        <div
          class="pb-2 flex justify-start"
        >
          <div>
            <div
              class="body2 pb-4"
            >
              <div>
                external_access.download_medical_records.disclaimer
              </div>
              <ul
                class="pl-4 py-2 list-disc"
              >
                <li>
                  external_access.download_medical_records.disclaimer_bullet_1
                </li>
                <li>
                  external_access.download_medical_records.disclaimer_bullet_2
                </li>
                <li>
                  external_access.download_medical_records.disclaimer_bullet_3
                </li>
              </ul>
              <div>
                external_access.download_medical_records.disclaimer_desc
              </div>
            </div>
            <button
              class="big-button w-full md:w-auto"
              data-test-id="submit-button"
              disabled=""
              type="submit"
            >
              <div
                class="flex flex-row justify-center items-center space-x-2"
              >
                <div>
                  external_access.download_medical_records.download
                </div>
              </div>
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="flex items-center justify-center m-auto justify-self-center"
    />
  </div>
</DocumentFragment>
`;
